Source/Dafny/AST/DafnyAst.cs

```diff
@@ -9070,6 +9086,48 @@ public abstract class Expression {
       get { yield break; }
     }
 
+    private IToken startTok;
+    private IToken endTok;
+
+    // Creates a token on the entire range of the expression.
+    // Used only for error reporting.
+    public IToken RangeToken => new RangeToken(StartTok, EndTok);
+
+    // Computes the start and end token of this expression
+    private void RecomputeStartEnd() {
```

Comment:
Why is this called `Re` compute, isn't it only called once?

Reply:
I imagine it could be called multiple times if we modify the AST, e.g. by adding elements in a set comprehension.